PyRDF/backend/AWS.py

```diff
@@ -92,20 +131,78 @@ def invoke_root_lambda(client, root_range, script):
                 'filelist': str(root_range.filelist),
                 'friend_info': encode_object(root_range.friend_info)
             })
-            return client.invoke(
-                FunctionName='root_lambda',
-                InvocationType='Event',
-                Payload=bytes(payload, encoding='utf8')
-            )
+
+            # Maybe here give info about number of invoked lambda for awsmonitor
+            #self.logger.info(f'New lambda - 31')
+
+            while trials:
+                trials -= 1
+                try:
+                    #my_before_time = time.time()
+                    response = client.invoke(
+                        FunctionName='root_lambda',
+                        InvocationType= 'Event', #'RequestResponse',
+                        Payload=bytes(payload, encoding='utf8')
+                    )
+                    #my_after_time = time.time()
+                    #self.logger.info(f'Invocation time: {my_after_time - my_before_time}')
+
+                    if 'FunctionError' in response:
+                        raise RuntimeError('Lambda runtime error')
+
+                    response = process_response(response)
+
+                    if 'Payload' in response:
+                        response_payload = response['Payload']
+                        lambda_status = response_payload['statusCode'] if 'statusCode' in response_payload else 200
```

Comment:
Can it ever not be in payload?

Reply:
Payload is a serialized value that lambda returns. By default it is dict with some values and one of the keys is statusCode, but it can be omitted. It was done without prior knowledge of lambda's insides and with an assumption that some lambda logic may be based on the early return of other statusCode f.e. to indicate error. This is not the case here, so can remove this whole paylod thing for a potential speedup.